cdviz-collector/src/sources/http.rs

```diff
@@ -46,13 +38,15 @@ pub(crate) struct HttpSource {
 #[derive(Clone)]
 struct AppState {
     tx: Sender<Message>,
+    event_type: Extension<EventType>,
```

Comment:
From my understanding, State is the successor of Extension.

In both case (State, Extension) the purpose is to share the same info between requests, handler. Are you sure (links) that Extension could be used to store information set by a middleware per request?